app/src/pages/HomePage.tsx

```diff
@@ -45,6 +65,15 @@ export const HomePage = (): React.ReactElement => {
                       imageUrl={image.imageUrl}
                     />
                   ))}
+                  {onImageFilesChosen ? (
```

Comment:
this looks wierd, why are you switching on the function? this should be that if the imageLink is populated, show the image otherwise should the chooser